x-pack/plugins/data_enhanced/public/search/search_interceptor.ts

```diff
@@ -84,18 +84,22 @@ export class EnhancedSearchInterceptor extends SearchInterceptor {
       if (id && !isSavedToBackground) this.deps.http.delete(`/internal/search/${strategy}/${id}`);
     });
 
-    return pollSearch(search, cancel, { ...options, abortSignal: combinedSignal }).pipe(
+    return pollSearch(search, cancel, {
+      ...options,
+      abortSignal: searchAbortController.getSignal(),
+    }).pipe(
       tap((response) => (id = response.id)),
       catchError((e: Error) => {
         cancel();
-        return throwError(this.handleSearchError(e, timeoutSignal, options));
+        return throwError(this.handleSearchError(e, options, searchAbortController.isTimeout()));
       }),
       finalize(() => {
         this.pendingCount$.next(this.pendingCount$.getValue() - 1);
-        cleanup();
-        if (untrackSearch && this.deps.session.isCurrentSession(options.sessionId)) {
+        if (this.deps.session.isCurrentSession(options.sessionId)) {
           // untrack if this search still belongs to current session
-          untrackSearch();
+          untrackSearch?.();
+        } else {
+          searchAbortController.cleanup();
```

Comment:
I'm not sure why we only have to clean up if it's not the current session... Can you explain?